src/sql/command.py

```diff
@@ -89,3 +88,30 @@ def connection(self):
     def result_var(self):
         """Returns the result_var"""
         return self.parsed["result_var"]
+
+    def _var_expand(self, magic, user_ns, line, cell):
+        """
+        Support for the variable substition in the SQL clause
+        For now, we have enabled two ways:
+        1. Latest format, {{a}}, we use jinja2 to parse the string with {{a}} format
+        2. Legacy format, {a}, $a, and :a format.
+
+        We will deprecate the legacy format feature in next major version
+        """
+        self.is_legacy_var_expand_parsed = False
+        # Latest format parsing
+        # TODO: support --param and --use-global logic here
+        # Ref: https://github.com/ploomber/jupysql/issues/93
+        line = Template(line).render(user_ns)
+        cell = Template(cell).render(user_ns)
+        # Legacy format parsing
+        parsed_cell = magic.shell.var_expand(cell, depth=2)
+        parsed_line = magic.shell.var_expand(line, depth=2)
+        # Exclusive the string with "://", but has :variable
+        has_SQLAlchemy_var_expand = re.search("(?<!://):[^/]+", line) or ":" in cell
```

Comment:
looks like we're mising warnings for {a} and $a?

Reply:
[This](https://github.com/ploomber/jupysql/pull/137/files#diff-769097f5e36be57ddec10c78df0900db8c1e254f5485bc94367aa7fcb07991c5R112) the line will check if throwing the warning for `{a} and $a`

`if parsed_line != line or parsed_cell != cell`, if the original sql clause is parsed by var_expand, that means the user uses either `{a} and $a`. We throw the warning by comparing the parsed and original

Reply:
Those test cases are covering
`test_variable_substitution_legacy_warning_message_single_curly`
`test_variable_substitution_legacy_warning_message_colon`